kitsune/questions/views.py

```diff
@@ -514,32 +506,20 @@ def aaq(
 
         return HttpResponseRedirect(path)
 
-    if product_key is None:
+    user_agent = request.META.get("HTTP_USER_AGENT", "")
+    is_mobile_device = parse_ua(user_agent).is_mobile
 
+    if product_key is None:
```

Comment:
You can remove this line and move the `product_key = request.GET.get("product")` to the same if statement below.

Reply:
We'll still need another `if product_key is None` somewhere so we don't overwrite a `?product=blah` on mobile. Thought this was cleaner than more nested ifs.

Thinking about it, IDK if `?product=blah` is actually used anywhere... but I don't want to open any more cans of worms before the responsive release.

Reply:
it's used only in mobile devices in order to avoid the UA check if the user decided to follow the `change product` link

Reply:
Lines 511 and 514 both have the same check. Why don't we add the line below under 514 so that we ca remove one of the two if?

Reply:
Because they're not the same check. They both check `product_key`, but line 512 modifies the value of `product_key`. So line 514 is actually `if product_key is None and request.GET.get("product") is None`.

Were we to move line 512 under 514 we'd still need another `if` to check if `request.GET.get("product")` is None.

Reply:
> it's used only in mobile devices in order to avoid the UA check if the user decided to follow the `change product` link

That's the `change_product` param used below: https://github.com/mozilla/kitsune/pull/4290/files#diff-f5ee25d927694fd302a3bb72ac3ebb4eR517

Reply:
something like `product_key or request.GET.get("product")`